packages/server/api/webscrape.py

```diff
@@ -3,32 +3,63 @@
 import re
 from urllib import parse
 from bs4 import BeautifulSoup
+import json
 
 class handler(BaseHTTPRequestHandler):
   whitelisted_url_regexes = [
     re.compile('^https:\/\/www\.bonappetit\.com')
   ]
 
+  def error_handler(error_string, error_code = 500):
+    self.send_response(error_code)
+    self.send_header('Content-type', 'text/plain')
+    self.end_headers()
+    self.wfile.write(errror_string.encode())
+
   def do_GET(self):
     url = parse.parse_qs(parse.urlsplit(self.path).query)['url'][0]
     url_in_whitelist = any(regex.match(url) for regex in self.whitelisted_url_regexes)
     
     if(not url_in_whitelist):
-      self.send_response(500)
-      self.send_header('Content-type', 'text/plain')
-      self.end_headers()
-      self.wfile.write('Unsupported recipe site'.encode())
+      error_handler('URL not supported')
+      return
+    
+    recipe_object = {
+      'servings': '',
+      'prepTime': '',
+      'cookTime': '',
+      'ingredients': '',
+      'steps': '',
+      'links': '',
+      'name': ''
+    }
+
+    try: 
+      file_pointer = request.urlopen(url)
+      html_bytes = file_pointer.read()
+      html_doc = html_bytes.decode('utf8')
+      file_pointer.close()
+      html_parsed = BeautifulSoup(html_doc, 'html.parser')
+    except:
+      error_handler('HTML could not be parsed')
       return
 
-    file_pointer = request.urlopen(url)
-    html_bytes = file_pointer.read()
-    html_doc = html_bytes.decode('utf8')
-    file_pointer.close()
-    html_parsed = BeautifulSoup(html_doc, 'html.parser')
-    ingredients = html_parsed.find('div', 'ingredients')
+    try: 
+      recipe_object['name'] = html_parsed.find('a', 'top-anchor').text
+      recipe_object['servings'] = html_parsed.find('span', 'recipe__header__servings recipe__header__servings--basically').span.text
+      ingredients = html_parsed.find_all('div', 'ingredients__text')
+      ingredients_array = list(map(lambda ingredient : { 'name': ingredient.text}, ingredients))
+      recipe_object['ingredients'] = ingredients_array
+      steps = html_parsed.find_all('li', 'step')
+      steps_array = list(map(lambda step : step.div.p.text, steps))
+      recipe_object['steps'] = steps_array
```

Comment:
I think this stuff should all be in its bon appetit specific function like
```python
def format_bon_appetit_recipe(html):

```

Reply:
Yeah we were going to abstract this out later on, should we make it part of this PR?

Reply:
We didn't do it right now because we didn't know the best approach, we didn't know if we put the function in a different file if that would convert it into a serverless function, etc. I think it would be best if we had a discussion about it and put it in another ticket

Reply:
I think. Since the implementation is specific to the Bon Appetit HTML, it should be in its own function

Reply:
https://www.notion.so/How-to-handle-ingredients-lists-that-don-t-have-separated-out-quantities-f1a05b7889bc497aa0281ad4836a7982